app/views/admin/api/new.html.erb

```diff
@@ -0,0 +1,14 @@
+<%= form_for @api_key, url: new_admin_api_path do |f| %>

```

Comment:
Like the edit page, this needs our standard page structure -- i.e. the admin div, a heading, and navigation. It also has some invalid HTML (a `<dl>` can only have `<dt>` and `<dd>` as its direct children, but this has a `<ul>`) and isn't quite set up like our usual forms, which are grouped into field sets with legends and headings that repeat the legend text (which is not the most efficient thing in the world, but screen readers don't always get along with the `<legend>` element). The way required fields are marked also needs some adjusting to be up to our new standards (which are undocumented, just like our old... well, "standards" would be overstating the matter).


```
<div class="admin">
  <!--Descriptive page name, messages and instructions-->
  <h2 class="heading"><%= ts("New API Token") %></h2>
  <!--/descriptions-->

  <!--subnav-->
  <ul class="navigation actions" role="navigation">
    <li><%= span_if_current ts("API Tokens"), admin_api_path %></li>
    <li><%= span_if_current("New API Token"), new_admin_api_path %></li>
  </ul>
  <!--/subnav-->

  <!--main content-->
  <%= form_for @api_key, url: admin_api_path(@api_key), html: { class: "post" } do |f| %>
    <p class="required notice">* <%= ts("Required information") %></p>
    <fieldset>
      <legend><%= ts("API Token") %></legend>
      <h3 class="landmark heading"><%= ts("API Token") %></h3>
      <dl>
        <dt class="required"><%= f.label :name, ts("Name") + "*" %></dt>
        <dd class="required">
          <%= f.text_field :name %>
          <%= live_validation_for_field("api_key_name", :presence => true %>
        </dd>
        <dt>
          <%= f.label :access_token, ts("Token (automatically generated)") %>
        </dt>
        <dd>
          <%= f.text_field :access_token, readonly: true %>
        </dd>
      </dl>
    </fieldset>
    <fieldset>
      <legend><%= ts("Actions") %></legend>
      <h3 class="landmark heading"><%= ts("Actions") %></h3>
      <p class="actions">
        <%= f.submit ts("Create API Token") %>
      </p>
    </fieldset>
  <% end %>
  <!--/content-->
</div>
```